src/relion/zocalo/wrapper.py

```diff
@@ -238,6 +238,11 @@ def start_relion(self):
         finally:
             os.chdir(oldpwd)
 
+        try:
+            icebreaker_histogram.create_histogram(self.working_directory)
+        except Exception as ex:
+            logger.error("Error creating Icebreaker histogram.", ex)
```

Comment:
I think the `icebreaker_histogram.create_histogram` code already catches the exceptions so they won't get caught here. Could remove the `try`/`except` from `create_histogram`?

Reply:
Yes, I thought about that, could I just have it return None instead?

Reply:
Yes, an indication of success through the return could be useful. I think what I would do is take the `try` out of `create_histogram` and put in a check on the existence of the `particles.star` file. Return `None` if it doesn't exist and the path to the pdf if it does. Then in the wrapper still have the `try`/`except` but only catch `RuntimeError` (for a gemmi problem) and maybe `ValueError`, which could come from some other file reading issue. Then the wrapper should just be catching race condition related problems

Reply:
Do you think I should change the logger messages to reflect what you've said, or leave them more general? I'm cautious about making them specific in case it causes more problems than it solves

Reply:
I think general is fine if the logger gets the traceback as well